central/reports/scheduler/v2/report_gen.go

```diff
@@ -0,0 +1,326 @@
+package v2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/reports/common"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/branding"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/mathutil"
+	"github.com/stackrox/rox/pkg/notifiers"
+	"github.com/stackrox/rox/pkg/retry"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/templates"
+	"github.com/stackrox/rox/pkg/timestamp"
+)
+
+func (s *scheduler) generateAndSendReport(req *ReportRequest, collection *storage.ResourceCollection) error {
+	err := s.upsertReportStatus(req.Ctx, req.ReportMetadata, storage.ReportStatus_PREPARING)
```

Comment:
Seems like the FAILURE state is not stored. Any error after changing the status to PREPARING is not moved to FAILURE within this function. Put the responsibility of state management in one place.

Reply:
The FAILURE state will be stored by `runSingleReport` func in reports/scheduler/v2/schedule.go . All funds in report_gen.go just deal with generating and sending the report. I guess I should also move storing SUCCESS state to `runSingleReport` for consistency.